prov/tcp/src/tcpx_eq.c

```diff
@@ -61,39 +78,60 @@ static struct fi_ops_eq tcpx_eq_ops = {
 	.strerror = ofi_eq_strerror,
 };
 
+static struct fi_ops tcpx_eq_fi_ops = {
+	.size = sizeof(struct fi_ops),
+	.close = tcpx_eq_close,
+	.bind = fi_no_bind,
+	.control = ofi_eq_control,
+	.ops_open = fi_no_ops_open,
+};
+
 int tcpx_eq_create(struct fid_fabric *fabric_fid, struct fi_eq_attr *attr,
 		   struct fid_eq **eq_fid, void *context)
 {
-	struct util_eq *eq;
+	struct tcpx_eq *eq;
 	struct fi_wait_attr wait_attr;
 	struct fid_wait *wait;
 	int ret;
 
-	ret = ofi_eq_create(fabric_fid, attr, eq_fid, context);
+	eq = calloc(1, sizeof(*eq));
+	if (!eq)
+		return -FI_ENOMEM;
+
+	ret = ofi_eq_init(fabric_fid, attr, &eq->util_eq.eq_fid, context);
 	if (ret) {
 		FI_WARN(&tcpx_prov, FI_LOG_EQ,
 			"EQ creation failed\n");
-		return ret;
+		goto err1;
 	}
 
-	eq = container_of(*eq_fid, struct util_eq, eq_fid);
-	eq->eq_fid.ops	= &tcpx_eq_ops;
+	ret = fastlock_init(&eq->lock);
+	if (ret)
+		goto err2;
```

Comment:
I think this should be goto err1.

Reply:
Yes. Err handling logic here is wrong. will fix it